lib/core/src/Cardano/Wallet/DB/Sqlite/CheckpointsOld.hs

```diff
@@ -225,19 +234,20 @@ mkStoreCheckpoints wid =
     Database operations
 -------------------------------------------------------------------------------}
 selectAllCheckpoints
-    :: forall s. PersistState s
+    :: forall s. PersistAddressBook s
     => W.WalletId
     -> SqlPersistT IO [(W.Slot, W.Wallet s)]
 selectAllCheckpoints wid = do
     cps <- fmap entityVal <$> selectList
         [ CheckpointWalletId ==. wid ]
         [ Desc CheckpointSlot ]
-    fmap catMaybes $ forM cps $ \cp -> do
+    -- FIXME LATER during ADP-1043: Presence of these tables?
+    prologue <- fromJust <$> loadPrologue wid
+    forM cps $ \cp -> do
         utxo <- selectUTxO cp
-        st <- selectState (checkpointId cp)
-        pure $
-            (\s -> let c = checkpointFromEntity @s cp utxo s in (getPoint c, c))
-            <$> st
+        discoveries <- loadDiscoveries wid (checkpointSlot cp)
+        let st = withIso addressIso $ \_ from -> from (prologue, discoveries)
+        pure $ let c = checkpointFromEntity @s cp utxo st in (getPoint c, c)
```

Comment:
would be in favor of extracting let in the above line. just for the sake of clarity

Reply:
Added in the additional commit.